### PR TITLE
layers: Replace searchs with device_profile_layer

### PR DESCRIPTION
**Pull request by jeremyk-lunarg**
Takes advantage of the device_profile_layer to replace searches in tests.

### CODE REVIEW

**Comment by jeremyk-lunarg** [comment]
@chrisforbes, does this look better and are there any other changes you would like me to make?

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -13045,11 +13062,19 @@ TEST_F(VkLayerTest, InvalidImageLayout) {
...
+    PFN_vkSetPhysicalDeviceFormatPropertiesEXT fpvkSetPhysicalDeviceFormatPropertiesEXT = nullptr;
+    PFN_vkGetOriginalPhysicalDeviceFormatPropertiesEXT fpvkGetOriginalPhysicalDeviceFormatPropertiesEXT = nullptr;
+
+    // Load required functions
+    if (!LoadDeviceProfileLayer(fpvkSetPhysicalDeviceFormatPropertiesEXT, fpvkGetOriginalPhysicalDeviceFormatPropertiesEXT)) {
+        printf("             device_profile_layer functions not found. Skipped.\n");
         return;
     }
+
+    depth_format = GetSupportedDepthStencilFormat(gpu(), fpvkSetPhysicalDeviceFormatPropertiesEXT, fpvkGetOriginalPhysicalDeviceFormatPropertiesEXT);
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: This is broken. A few lines later when you create the image, you end up passing VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT down to the driver for this format. At that point you have undefined behavior.

**Comment by jeremyk-lunarg** [comment]
@chrisforbes, I fixed it, but it looks like all that this changes now is adding the function GetSupportedDepthStencilFormat which is never used.

**Comment by chrisforbes** [comment]
Closing for now, then. Not opposed to the general idea if you find cases where it simplifies /and/ is safe.

**Pull request closed without merging**